common/src/java/org/apache/hadoop/hive/common/type/Date.java

```diff
@@ -177,13 +177,22 @@ public void setTimeInMillis(long epochMilli) {
    * @throws NullPointerException if {@code text} is null
    */
   public static Date valueOf(final String text) {
-    String s = Objects.requireNonNull(text).trim();
+    String trimedText = Objects.requireNonNull(text).trim();
     ParsePosition pos = new ParsePosition(0);
     try {
-      TemporalAccessor t = PARSE_FORMATTER.parseUnresolved(s, pos);
+      TemporalAccessor t = PARSE_FORMATTER.parse(trimedText, pos);
       if (pos.getErrorIndex() >= 0) {
-        throw new DateTimeParseException("Text could not be parsed to date", s, pos.getErrorIndex());
+        throw new DateTimeParseException("Text could not be parsed to date", trimedText, pos.getErrorIndex());
       }
+      //Check if there is still text left after parsing
+      if(pos.getIndex() < trimedText.length()) {
+        char lastChar = trimedText.charAt(pos.getIndex());
+        //Check if the first character of the remaining is a digit, e.g. "2023-08-0800" if yes, then it is a parse error and must throw an exception
+        if (lastChar >= '0' && lastChar <= '9'){
+          throw new DateTimeParseException("Text '" + trimedText + "' could not be parsed, unparsed text found at index " + pos.getIndex(), trimedText,
+              pos.getIndex());
+        }
+      }  
```

Comment:
This logic could be a little naive but I don't come up with a better solution

Reply:
I tried to find better solutions, but this was the most effective and clear option.